src/ert/_c_wrappers/enkf/enkf_obs.py

```diff
@@ -144,7 +148,300 @@ def addObservationVector(self, observation_vector: ObsVector) -> None:
+                        warnings.warn(
+                            f"Segment {segment_instance.name} does not"
+                            " contain any time steps. The interval "
+                            f"[{start}, {stop}) does not intersect with steps in the"
+                            "time map.",
+                            category=ConfigWarning,
+                        )
+                    std_dev[start:stop] = self._handle_error_mode(
+                        values[start:stop],
+                        float(segment_instance.get_value("ERROR")),
+                        float(segment_instance.get_value("ERROR_MIN")),
+                        segment_instance.get_value("ERROR_MODE"),
+                    )
+                for i, (good, error, value) in enumerate(zip(valid, std_dev, values)):
+                    if good:
+                        if error <= std_cutoff:
+                            warnings.warn(
+                                "Too small observation error in observation"
+                                f" {summary_key}:{i} - ignored",
+                                category=ConfigWarning,
+                            )
+                            continue
+                        obs_vector.add_summary_obs(
+                            SummaryObservation(summary_key, summary_key, value, error),
+                            i,
+                        )
+
+            self.addObservationVector(obs_vector)
+
+    @staticmethod
+    def _get_time(conf_instance, start_time: datetime) -> Tuple[datetime, str]:
+        if conf_instance.has_value("DATE"):
+            date_str = conf_instance.get_value("DATE")
+            try:
+                return datetime.fromisoformat(date_str), f"DATE={date_str}"
+            except ValueError:
+                try:
+                    date = datetime.strptime(date_str, "%d/%m/%Y")
+                    warnings.warn(
+                        f"Deprecated time format {date_str}."
+                        " Please use ISO date format YYYY-MM-DD",
+                        category=ConfigWarning,
+                    )
+                    return date, f"DATE={date_str}"
+                except ValueError as err:
+                    raise ValueError(
+                        f"Unsupported date format {date_str}."
+                        " Please use ISO date format"
+                    ) from err
+
+        if conf_instance.has_value("DAYS"):
+            days = float(conf_instance.get_value("DAYS"))
+            return start_time + timedelta(days=days), f"DAYS={days}"
+        if conf_instance.has_value("HOURS"):
+            hours = float(conf_instance.get_value("HOURS"))
+            return start_time + timedelta(hours=hours), f"HOURS={hours}"
+        raise ValueError("Missing time specifier")
+
+    @staticmethod
+    def _find_nearest(
+        time_map: List[datetime],
+        time: datetime,
+        threshold: timedelta = timedelta(seconds=30),
```

Comment:
Just specify as None, and set during call if none?

github-lint: Do not perform function calls in argument defaults. The call is performed only once at function definition time. All calls to your function will reuse the result of that definition-time function call. If this is intended, assign the function call to a module-level variable and use that variable as a default value.